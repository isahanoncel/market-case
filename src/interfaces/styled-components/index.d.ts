// #region Global Imports
import 'styled-components';
// #endregion Global Imports

export type Icons = 'Bag' | 'ArrowLeft';

export type Colors =
  | 'primary'
  | 'secondary'
  | 'white'
  | 'black100'
  | 'black600';

type Fonts = 'OpenSans';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: Record<Colors, string>;
    fonts: Record<Fonts, string>;
  }
}
