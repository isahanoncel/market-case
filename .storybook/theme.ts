import { create } from '@storybook/theming';

export default create({
  base: 'light',
  brandTitle: 'Getir',
  brandUrl: 'https://getir.com',
  brandTarget: '_self',
});
