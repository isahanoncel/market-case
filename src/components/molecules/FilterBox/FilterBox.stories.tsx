import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import FilterBox from '.';

export default {
  title: 'Molecules/FilterBox',
  component: FilterBox,
  args: {
    title: 'Brands',
    items: [
      { key: 'key1', value: 'value 1' },
      { key: 'key2', value: 'value 2' },
    ],
  },
} as ComponentMeta<typeof FilterBox>;

export const Default: ComponentStory<typeof FilterBox> = (args) => (
  <FilterBox {...args} />
);
