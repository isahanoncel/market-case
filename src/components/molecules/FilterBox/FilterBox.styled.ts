import styled from 'styled-components';
import { device } from '../../../constants/styles/GlobalStyle';
import StyledHelper from '../../../constants/styles/Helper';

export const FilterBoxWrapper = styled.div`
  background-color: ${({ theme }) => theme.colors.white};
  box-shadow: 0px 6px 24px rgba(93, 62, 188, 0.04);
  border-radius: 2px;
  margin-top: 12px;
  padding: 24px;
  max-height: 244px;
  max-width: 296px;
  overflow: hidden;
  ${device.mobile} {
    max-width: 100%;
  }
`;

export const FilterListWrapper = styled.div`
  max-height: 131px;
  overflow-y: scroll;
`;

export const FilterItem = styled.div`
  cursor: pointer;
  margin-bottom: 18px;
  ${StyledHelper.CenterContentVertical}

  &:last-of-type {
    margin-bottom: 0;
  }
`;

export const FilterCheck = styled.div<{ checked?: boolean }>`
  ${StyledHelper.CenterContent}
  width: 22px;
  height: 22px;

  box-shadow: 0px 1px 7px rgba(93, 56, 192, 0.4);
  border-radius: 2px;

  margin: 0 8px 0 0;
  ${(props) =>
    props.checked && `background-color:${props.theme.colors.primary}`};
`;
