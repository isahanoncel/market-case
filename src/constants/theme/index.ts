import { DefaultTheme } from 'styled-components';

export const COLORS = {
  white: '#fff',
  primary: '#1EA4CE',
  whiteBlue: '#F2F0FD',
  secondary: '#147594',
  black100: '#E0E0E0',
  black500: '#6F6F6F',
  black600: '#525252',
  gray: '#697488',
  background: '#E5E5E5',
  transparent: 'transparent',
  innerBorder: '#DFDEE2',
  thinGray: '#A8A8A8',
};

export const FONTS = {
  OpenSans: "'Open Sans', sans-serif",
};

const heroTheme: DefaultTheme = {
  colors: COLORS,
  fonts: FONTS,
};

export default heroTheme;
