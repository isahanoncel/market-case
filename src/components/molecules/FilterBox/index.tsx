import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
import { IOption } from '../../../interfaces/option';
import { useAppDispatch, useAppSelector } from '../../../store';
import { setBrands, setTags } from '../../../store/slices/filter';
import Flex from '../../atoms/Flex';
import Input from '../../atoms/Input';
import Label from '../../atoms/Label';
import * as S from './FilterBox.styled';

interface IFilterBox {
  title: string;
  items: IOption[];
  type?: string;
}

const FilterBox: FC<IFilterBox> = ({ title, items, type }) => {
  const [filteredItems, setFilteredItems] = useState<IOption[]>(items);
  const [keyword, setKeyword] = useState<string>('');
  const brands = useAppSelector((a) => a.filter.brands);
  const tags = useAppSelector((a) => a.filter.tags);
  const dispatch = useAppDispatch();

  useEffect(() => {
    setFilteredItems(
      items.filter((item) =>
        item.key.toLowerCase().includes(keyword.toLowerCase()),
      ),
    );
  }, [keyword, items, brands, tags]);

  const handleSetFilterItem = useCallback((data: string) => {
    if (type === 'tag') {
      dispatch(setTags(data));
      return;
    }

    dispatch(setBrands(data));
  }, []);

  const renderItems = useMemo(() => {
    return filteredItems.map((item) => {
      const isItemChecked = (type === 'tag' ? tags : brands)?.includes(
        item.value,
      );
      return (
        <S.FilterItem
          key={item.key}
          onClick={() => handleSetFilterItem(item.value)}
        >
          <S.FilterCheck checked={isItemChecked}>
            {isItemChecked && (
              <svg
                width="10"
                height="7"
                viewBox="0 0 10 7"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 1L3.5 6L1 3.72727"
                  stroke="#FFF"
                  strokeWidth="1.2"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />
              </svg>
            )}
          </S.FilterCheck>
          <Flex alignItems="center">
            <Label color="black600" weight={600} size={14}>
              {item.key}
            </Label>

            <Label color="thinGray" size={14} margin="0 0 0 4px">
              (18)
            </Label>
          </Flex>
        </S.FilterItem>
      );
    });
  }, [filteredItems]);

  return (
    <Flex flexDirection="column" margin="24px 0 0 0">
      <Label size={13} weight={600} color="gray">
        {title}
      </Label>

      <S.FilterBoxWrapper>
        <Input
          onChange={(text) => setKeyword(text)}
          placeholder={`Search ${title}`}
        />
        <S.FilterListWrapper>{renderItems}</S.FilterListWrapper>
      </S.FilterBoxWrapper>
    </Flex>
  );
};

export default FilterBox;
