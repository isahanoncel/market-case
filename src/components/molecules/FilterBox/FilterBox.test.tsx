import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import FilterBox from '.';
import theme from '../../../constants/theme';

test('FilterBox snapshot testing', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <FilterBox
          items={[
            { key: 'key1', value: 'value 1' },
            { key: 'key2', value: 'value 2' },
          ]}
          title="Brands"
        />
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
