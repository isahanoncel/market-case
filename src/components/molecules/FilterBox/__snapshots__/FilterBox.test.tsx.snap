// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBox snapshot testing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 24px 0 0 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  margin-bottom: 17px;
}

.c5 {
  font-size: 14px;
  outline: none;
  padding: 12px 16px;
  color: #525252;
  border: 2px solid #E0E0E0;
  border-radius: 2px;
}

.c5::-webkit-input-placeholder {
  color: #E0E0E0;
}

.c5::-moz-placeholder {
  color: #E0E0E0;
}

.c5:-ms-input-placeholder {
  color: #E0E0E0;
}

.c5::placeholder {
  color: #E0E0E0;
}

.c1 {
  font-size: 13px;
  line-height: 14px;
  color: #697488;
  font-weight: 600;
  margin: 0;
}

.c10 {
  font-size: 14px;
  line-height: 14px;
  color: #525252;
  font-weight: 600;
  margin: 0;
}

.c11 {
  font-size: 14px;
  line-height: 14px;
  color: #A8A8A8;
  font-weight: 400;
  margin: 0 0 0 4px;
}

.c2 {
  background-color: #fff;
  box-shadow: 0px 6px 24px rgba(93,62,188,0.04);
  border-radius: 2px;
  margin-top: 12px;
  padding: 24px;
  max-height: 244px;
  max-width: 296px;
  overflow: hidden;
}

.c6 {
  max-height: 131px;
  overflow-y: scroll;
  padding: 2px 4px;
}

.c7 {
  cursor: pointer;
  margin-bottom: 18px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7:last-of-type {
  margin-bottom: 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 22px;
  height: 22px;
  box-shadow: 0px 1px 7px rgba(93,56,192,0.4);
  border-radius: 2px;
  margin: 0 8px 0 0;
}

@media (min-width:0px) and (max-width:768px) {
  .c0 {
    height: auto;
  }
}

@media (min-width:0px) and (max-width:768px) {
  .c4 {
    height: auto;
  }
}

@media (min-width:0px) and (max-width:768px) {
  .c9 {
    height: auto;
  }
}

@media (min-width:0px) and (max-width:768px) {
  .c2 {
    max-width: 100%;
  }
}

<div
  className="c0"
>
  <span
    className="c1"
    color="gray"
    size={13}
  >
    Brands
  </span>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <input
          className="c5"
          onChange={[Function]}
          placeholder="Search Brands"
          type="text"
        />
      </div>
    </div>
    <div
      className="c6"
    >
      <div
        className="c7"
        onClick={[Function]}
      >
        <div
          checked={false}
          className="c8"
        />
        <div
          className="c9"
        >
          <span
            className="c10"
            color="black600"
            size={14}
          >
            key1
          </span>
          <span
            className="c11"
            color="thinGray"
            size={14}
          />
        </div>
      </div>
      <div
        className="c7"
        onClick={[Function]}
      >
        <div
          checked={false}
          className="c8"
        />
        <div
          className="c9"
        >
          <span
            className="c10"
            color="black600"
            size={14}
          >
            key2
          </span>
          <span
            className="c11"
            color="thinGray"
            size={14}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
